actions/webhook.go

```diff
@@ -33,32 +34,39 @@ type WebhookEventInfo struct {
 }
 
 const webhookClientTimeout = 5 * time.Minute
+const queryParamsPropertyKey = "query_params"
 
 var (
 	ErrWebhookRequestFailed = errors.New("webhook request failed")
-	ErrWebhookMissingURL    = errors.New("webhook missing url")
+	ErrWebhookWrongFormat   = errors.New("webhook wrong format")
 )
 
 func NewWebhook(h ActionHook, action *Action) (Hook, error) {
-	webhookURL := h.Properties["url"]
+	webhookURL := h.Properties["url"].(string)
 	if len(webhookURL) == 0 {
-		return nil, ErrWebhookMissingURL
+		return nil, fmt.Errorf("missing url: %w", ErrWebhookWrongFormat)
+	}
+
+	queryParams, err := extractQueryParams(h.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("extracting query params: %w", err)
 	}
 
 	requestTimeout := webhookClientTimeout
-	timeoutDuration := h.Properties["timeout"]
-	if len(timeoutDuration) > 0 {
-		d, err := time.ParseDuration(timeoutDuration)
+	timeoutDuration, ok := h.Properties["timeout"]
+	if ok && len(timeoutDuration.(string)) > 0 {
```

Comment:
You will need to do this one first:
```go
timeoutString, ok := timeoutDuration.(string)
```
to verify the property is a string - without the explicit capture you will get panic